lib/tmuxinator/hooks.rb

```diff
@@ -0,0 +1,39 @@
+module Tmuxinator
+  module Hooks
+    def hook_on_reattach
```

Comment:
I didn't propose such solution because I preferred to give something simpler to review as is my first contribution, but if that is the preferred path for this feature I'll be more than welcome in contributing on that!

Also be aware that this is a "hook on tmuxinator project events" and not a "hook on tmux event".  
AFAIU tmux hooks are global/session level configuration, while this PR simply expand the `pre` and `post` configurations already available.

I probably choose the wrong name "hooks" as is misleading in this context, and talking about *tmux session events* helped the misunderstanding. I'm sorry. I reworded the title to be more correct.